src/testUtils.ts

```diff
@@ -224,6 +263,21 @@ export function showTestOutput() {
 	outputChannel.show(true);
 }
 
+/**
+ * Iterates the list of currently running test processes and kills them all.
+ */
+export function cancelRunningTests(): Thenable<boolean> {
+	return new Promise<boolean>((resolve, reject) => {
+		testProcesses.forEach(function(tp){
+			tp.kill(sendSignal);
+		});
+		// All processes are now dead. Empty the array to prepare for the next run.
+		testProcesses.splice(0, testProcesses.length);
```

Comment:
Killed test processes go through like 239 where they are removed from the `testProcesses` array. Do we need to empty the array again here?

Reply:
In this case, I do not know. I don't know if the kills are done in the background while the `cancelRunningTests` function continues to run. I wrote it such that we would have a clean state were that the case. Is that the case? I know little about the runtime here.

Reply:
I am not really sure..
But, there is no harm emptying the array at this point so we can keep it